src/components/OfferSwitchCard/OfferSwitchCard.tsx

```diff
@@ -48,7 +48,7 @@ const OfferSwitchCard = ({ baseOfferId, toOfferId }: OfferSwitchCardProps) => {
         CurrencyFormat,
         string
       >
-    ];
+    ] || switchDetails?.nextPaymentPriceCurrency;
```

Comment:
this might still throw an `undefined`

Reply:
@iwonakulacz yes, but what the nullish coalescing operator does is that if the `switchDetails` does not exist, the whole expression will return `undefined`. which is, what i understood, what we wanted to avoid, right?